assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -244,7 +252,7 @@ class AssistedInjectProcessor : AbstractProcessor() {
       throw StopProcessingException("Factory interface defines multiple factory methods.", factory)
     }
     val method = methods.single()
-    if (!types.isAssignable(type.asType(), method.returnType)) {
+    if (!types.isAssignable(types.erasure(type.asType()), types.erasure(method.returnType))) {
```

Comment:
i agree, and it's really confusing why it didn't work. 

<img width="483" alt="screen shot 2017-07-18 at 2 35 17 pm" src="https://user-images.githubusercontent.com/1675456/28333587-5acba96a-6bc6-11e7-834d-c73b4958490d.png">